File: app/src/main/java/igrek/todotree/service/history/change/ModifyItemItemChange.java
package igrek.todotree.service.history.change;


import igrek.todotree.domain.treeitem.AbstractTreeItem;

public class ModifyItemItemChange extends AbstractItemChange {
	
	private AbstractTreeItem modifiedItem;
	private String oldContent;
	private String newContent;
	
	public ModifyItemItemChange(AbstractTreeItem modifiedItem, String oldContent, String newContent) {
		this.modifiedItem = modifiedItem;
		this.oldContent = oldContent;
		this.newContent = newContent;
	}
	
	public AbstractTreeItem getModifiedItem() {
		return modifiedItem;
	}
	
	public String getOldContent() {
		return oldContent;
	}
	
	public String getNewContent() {
		return newContent;
	}
	
	@Override
	public void revert() {
		//TODO
	}
}

File: app/src/main/java/igrek/todotree/service/history/change/ModifyItemItemChange.kt
package igrek.todotree.service.history.change

import igrek.todotree.domain.treeitem.AbstractTreeItem
import igrek.todotree.service.history.change.AbstractItemChange

class ModifyItemItemChange(
    val modifiedItem: AbstractTreeItem,
    val oldContent: String,
    val newContent: String
) : AbstractItemChange() {
    override fun revert() {
        //TODO
    }
}